OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitYoutubeVideoFragment.java

```diff
@@ -0,0 +1,436 @@
+package org.edx.mobile.view;
+
+import android.content.pm.ActivityInfo;
+import android.content.res.Configuration;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.v4.app.Fragment;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.Toast;
+
+
+import org.edx.mobile.R;
+import org.edx.mobile.base.BaseFragmentActivity;
+import org.edx.mobile.model.api.TranscriptModel;
+import org.edx.mobile.model.course.VideoBlockModel;
+import org.edx.mobile.model.db.DownloadEntry;
+import org.edx.mobile.module.db.impl.DatabaseFactory;
+import org.edx.mobile.player.TranscriptManager;
+import org.edx.mobile.util.NetworkUtil;
+import org.edx.mobile.util.VideoUtil;
+
+
+import com.google.android.youtube.player.YouTubePlayer;
+import com.google.android.youtube.player.YouTubePlayer.Provider;
+import com.google.android.youtube.player.YouTubeInitializationResult;
+import com.google.android.youtube.player.YouTubePlayerSupportFragment;
+
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Locale;
+
+import subtitleFile.Caption;
+import subtitleFile.FormatSRT;
+import subtitleFile.TimedTextObject;
+
+public class CourseUnitYoutubeVideoFragment extends CourseUnitVideoFragment implements YouTubePlayer.OnInitializedListener {
+
+    private static final int SUBTITLES_DISPLAY_DELAY_MS = 50;
+
+    private YouTubePlayer youTubePlayer, previousYouTubePlayer;
+    private Handler subtitleDisplayHandler = new Handler();
+    private Handler transcriptsHandler = new Handler();
+    private Handler initializeHandler = new Handler();
+    private TimedTextObject subtitlesObj;
+    private LinkedHashMap<String, TimedTextObject> srtList = new LinkedHashMap<>();
+    private YouTubePlayerSupportFragment youTubePlayerFragment;
+    private boolean fromLandscape, wasHiding;
+    private int attempts;
+
+    /**
+     * Create a new instance of fragment
+     */
+
+    public static CourseUnitYoutubeVideoFragment newInstance(VideoBlockModel unit, boolean hasNextUnit, boolean hasPreviousUnit) {
+        CourseUnitYoutubeVideoFragment fragment = new CourseUnitYoutubeVideoFragment();
+        fragment.setArguments(getCourseUnitBundle(unit, hasNextUnit, hasPreviousUnit));
+        return fragment;
+    }
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        setHasOptionsMenu(true);
+        releaseYoutubePlayer();
+        addOrRemoveYoutubePlayerFragment(VideoUtil.isAPIYoutubeSupported(getContext()));
+        attempts = 0;
+    }
+
+    @Override
+    public void setUserVisibleHint(boolean isVisibleToUser) {
+        super.setUserVisibleHint(isVisibleToUser);
+        releaseYoutubePlayer();
```

Comment:
IMO we should `releaseYoutubePlayer()` in else part means when `isVisibleToUser` is `false`.